modules/mcuboot/CMakeLists.txt

```
@@ -71,6 +72,10 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
       set(${SIGN_ARG_SIGNED_HEX_TEST_FILE_OUT} ${test_update_hex} PARENT_SCOPE)
     endif()
 
+    if(DEFINED SIGN_ARG_ROM_FIXED)
+      set(rom_fixed_option --rom-fixed ${SIGN_ARG_ROM_FIXED})
```

as mentioned in other comment, this only sets the var if `SIGN_ARG_ROM_FIXED` is defined, but in cases where it is not defined then the value of `rom_fixed_option` is picked up from the parent scope.

One option is to ensure the local scope var is always unset, but a another approach can simply be to prepend the arg var with the required prefix, like:
```suggestion
      set(SIGN_ARG_ROM_FIXED --rom-fixed ${SIGN_ARG_ROM_FIXED})
```